tools/plugins/com.liferay.ide.project.ui/src/com/liferay/ide/project/ui/migration/MigrationViewTreeUtil.java

```diff
@@ -0,0 +1,193 @@
+/*******************************************************************************
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ *******************************************************************************/
+
+package com.liferay.ide.project.ui.migration;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.jface.viewers.ITreeContentProvider;
+import org.eclipse.ui.navigator.CommonViewer;
+
+/**
+ * @author Terry Jia
+ */
+public class MigrationViewTreeUtil
+{
+
+    private final String CONTENT_PROVIDER_ID = "com.liferay.ide.project.ui.migration.content";
+
+    private CommonViewer _commonViewer;
+
+    private List<MPNode> _treeList;
```

Comment:
These two fields need to not be cached here.  Please resend with a new solution that doesn't cache.